File: Java/TP2-3/src/Celular.java

public class Celular {
	private int numero;
	private String usuario;
	private Bateria bateria;
	private boolean estado;

	public Celular(int numero, String usuario, Bateria bateria) {
		this.numero = numero;
		this.usuario = usuario;
		this.bateria = bateria;
		this.estado = false;
	}

	public void ligar() {
		if (this.estado == false) {
			if (bateria.getCarga() > 20) {
				System.out.println("Inicializando Android 7.0 " + this.usuario);
			} else if (this.bateria.getCarga() < 20 && this.bateria.getCarga() > 0) {
				System.out.println("Bateria Fraca");
			}
			this.estado = true;
			bateria.descarregar();
			bateria.descarregar();
		}
	}

	public void desligar() {
		if (this.estado == true) {
			System.out.println("Desligando...");
			bateria.descarregar();
			this.estado = false;
		}
	}

	public void musica() {
		if (this.estado = true && bateria.getCarga() >= 10) {
			System.out.println("Music Audio");
		}
	}

	public void trocarBateria(Bateria bateria) {
		System.out.println("Mudando Bateria");
		desligar();
		this.bateria = bateria;
	}

	public int getNumero() {
		return numero;
	}

	public String getUsuario() {
		return usuario;
	}

	public boolean isEstado() {
		return estado;
	}

	public Bateria getBateria() {
		return bateria;
	}

	public void mostraCarga() {
		this.bateria.mostraCarga();
	}

}

File: Kotlin/TPK2-3/bin/Celular.kt
import java.util.*

class Celular(numero: Int, usuario: String, bateria: Bateria) {
	private var numero: Int = 0
	private var usuario: String
	private var bateria: Bateria
	private var Estado: Boolean = false

	init {
		this.numero = numero
		this.usuario = usuario
		this.bateria = bateria
		this.Estado = false
	}

	fun ligar() {
		if (this.Estado == false) {
			if (bateria.Carga() > 20) {
				println("Inicializando Android 7.0 " + this.usuario)
			} else if (this.bateria.Carga() < 20 && this.bateria.Carga() > 0) {
				println("Bateria Fraca... favor recarregar")
			}
			this.Estado = true
			bateria.descarregar()
			bateria.descarregar()
		}
	}

	fun desligar() {
		if (this.Estado == true) {
			println("Desligando...")
			bateria.descarregar()
			this.Estado = false
		}
	}

	fun musica() {
		if (this.Estado == true && this.bateria.Carga() > 20) {
			println("Music Audio")
			bateria.descarregar()
		}else
			println("Bateria Fraca...")
	}

	fun trocarBateria(bateria: Bateria) {
		println("Mudando Bateria")
		desligar()
		this.bateria = bateria
	}

	fun mostraCarga() {
		this.bateria.mostraCarga()
	}
}